src/Modules/Settings/Dnn.PersonaBar.Prompt/Services/CommandController.cs

```diff
@@ -65,24 +65,17 @@ private new PortalSettings PortalSettings
         public HttpResponseMessage Cmd(int portalId, [FromBody] CommandInputModel command)
         {
             PortalId = portalId;
-            var portal = PortalController.Instance.GetPortal(PortalId);
+            var portal = PortalController.Instance.GetPortalSettings(portalId);
+
             if (portal == null)
             {
                 var errorMessage = string.Format(Localization.GetString("Prompt_GetPortal_NotFound", Constants.LocalResourcesFile), portalId);
                 Logger.Error(errorMessage);
                 return AddLogAndReturnResponse(null, null, command, DateTime.Now, errorMessage);
             }
 
-            if (portalId != base.PortalId)
-            {
-                if (!PortalHelper.IsRequestForSiteGroup(portalId, base.PortalSettings.PortalId))
-                {
-                    var errorMessage = string.Format(Localization.GetString("InValidPortalId", Constants.LocalResourcesFile), portalId);
-                    Logger.Error(errorMessage);
-                    return AddLogAndReturnResponse(null, null, command, DateTime.Now, errorMessage);
-                }
-                PortalSettings = new PortalSettings(PortalId);
-            }
+            PortalSettings = new PortalSettings(PortalId);
```

Comment:
portal id check not taken in account any more?

Reply:
This was much needed to differentiate in rest of life cycle that either we have different PortalSettings object than the settings obtained from PortalController.Instance.GetCurrentPortalSettings() which is by default for request.